File: app/src/main/java/co/tpcreative/supersafe/common/controller/SingletonFakePinComponent.java
package co.tpcreative.supersafe.common.controller;

import co.tpcreative.supersafe.common.util.Utils;

public class SingletonFakePinComponent {

    private static SingletonFakePinComponent instance;
    private SingletonPrivateFragmentListener ls;

    public static SingletonFakePinComponent getInstance() {
        if (instance==null){
            instance = new SingletonFakePinComponent();
        }
        return instance;
    }

    public void setListener(SingletonPrivateFragmentListener ls){
        this.ls = ls;
    }

    public void onUpdateView(){
        if (Utils.isLockedScreen()){
            return;
        }
        if (ls!=null){
            ls.onUpdateView();
        }
    }

    public interface SingletonPrivateFragmentListener{
        void onUpdateView();
    }

}

File: app/src/main/java/co/tpcreative/supersafe/common/controller/SingletonFakePinComponent.kt
package co.tpcreative.supersafe.common.controller

class SingletonFakePinComponent {
    private var ls: SingletonPrivateFragmentListener? = null
    fun setListener(ls: SingletonPrivateFragmentListener?) {
        this.ls = ls
    }

    fun onUpdateView() {
        if (SingletonManager.getInstance().isVisitLockScreen()) {
            return
        }
        ls.let {
            it?.onUpdateView()
        }
    }

    interface SingletonPrivateFragmentListener {
        fun onUpdateView()
    }

    companion object {
        private var instance: SingletonFakePinComponent? = null
        fun getInstance(): SingletonFakePinComponent {
            if (instance == null) {
                instance = SingletonFakePinComponent()
            }
            return instance!!
        }
    }
}